pkg/virt-handler/vm_test.go

```diff
@@ -3254,6 +3318,58 @@ var _ = Describe("DomainNotifyServerRestarts", func() {
 	})
 })
 
+var _ = Describe("IsoGuestVolumePath", func() {
+	table.DescribeTable("volume path for", func(volume *v1.Volume, expectedVolumePath string, expectedResult bool) {
+		vmi := v1.NewMinimalVMI("testvmi")
+		volumePath, result := IsoGuestVolumePath(vmi, volume)
+		Expect(result).To(BeEquivalentTo(expectedResult))
+		Expect(volumePath).To(BeEquivalentTo(expectedVolumePath))
+	},
+		table.Entry("cloudInitNoCloud", &v1.Volume{
+			VolumeSource: v1.VolumeSource{
+				CloudInitNoCloud: &v1.CloudInitNoCloudSource{},
+			},
+		}, "/var/run/kubevirt-ephemeral-disks/cloud-init-data/default/testvmi/noCloud.iso", true),
+		table.Entry("cloudInitConfigDrive", &v1.Volume{
+			VolumeSource: v1.VolumeSource{
+				CloudInitConfigDrive: &v1.CloudInitConfigDriveSource{},
+			},
+		}, "/var/run/kubevirt-ephemeral-disks/cloud-init-data/default/testvmi/configdrive.iso", true),
+		table.Entry("configMap", &v1.Volume{
+			Name: "test",
+			VolumeSource: v1.VolumeSource{
+				ConfigMap: &v1.ConfigMapVolumeSource{},
+			},
+		}, "/var/run/kubevirt-private/config-map-disks/test.iso", true),
+		table.Entry("downwardApi", &v1.Volume{
+			Name: "test",
+			VolumeSource: v1.VolumeSource{
+				DownwardAPI: &v1.DownwardAPIVolumeSource{},
+			},
+		}, "/var/run/kubevirt-private/downwardapi-disks/test.iso", true),
+		table.Entry("secret", &v1.Volume{
+			Name: "test",
+			VolumeSource: v1.VolumeSource{
+				Secret: &v1.SecretVolumeSource{},
+			},
+		}, "/var/run/kubevirt-private/secret-disks/test.iso", true),
+		table.Entry("serviceAccount", &v1.Volume{
+			VolumeSource: v1.VolumeSource{
+				ServiceAccount: &v1.ServiceAccountVolumeSource{},
+			},
+		}, "/var/run/kubevirt-private/service-account-disk/service-account.iso", true),
+		table.Entry("sysprep", &v1.Volume{
+			Name: "test",
+			VolumeSource: v1.VolumeSource{
```

Comment:
You could reduce the scope even to the `VolumeSource` as parameter and construct the VM in the test to make it even more focused.